pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -74,6 +70,20 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 					if currentCluster.DeletionTimestamp != nil {
 						return
 					}
+
+					// See GitHub issue #194: Migrate Client().Update() calls
+					// back to Patch() calls.
```

Comment:
Since we're now using a subresource for Status, I don't think the patch issue is relevant here.

Or that a resourceVersion conflict is possible (since no one other than us can update the status)? Although I guess I'm just assuming that the main-resource resourceVersion is irrelevant for the subresource. Have you observed a resourceVersion conflict here or is this block just being cautious?

Reply:
The ResourceVersion field is shared between the Spec or Status and is updated whenever either is modified. I have observed the for loop failing forever so checking for the conflict error here is necessary.

Reply:
Got it. OK, the conflict check is necessary then, but I think the comment about using PATCH is not necessary here.

Reply:
Even though we're not calling Update() here I think the comment is still necessary because when we go back to using Patch() I don't think we'll get the conflict error so the conflict check can be removed. Right?

Reply:
I would think we would keep using Status().Update() here because we'll still be updating the entire status subresource. (We can't patch the status clause of the main resource because that will be ignored, given the existence of the subresource.)